app/workers/publishing_api_document_republishing_worker.rb

```diff
@@ -12,46 +12,80 @@
-    document = Document.find(document_id)
 
-    # this the latest edition in a visible state ie: withdrawn, published
-    @live_edition = document.live_edition
-
-    @pre_publication_edition = document.pre_publication_edition
-    @latest_unpublished_edition = document.editions.unpublished.last
+    set_document_and_editions(document_id)
 
     return unless the_document_has_non_superseded_editions_to_republish?
 
     Document.transaction do
       document.lock!
 
-      if the_document_has_been_unpublished?
-        send_unpublish(latest_unpublished_edition)
-        send_draft_edition if pre_publication_edition.present?
-      elsif the_document_has_been_withdrawn?
-        send_published_and_withdraw
-      elsif there_is_only_a_draft?
-        patch_links
-        send_draft_edition
-      elsif there_is_only_a_live_edition?
-        patch_links
-        send_live_edition
-      elsif there_is_a_newer_draft?
-        patch_links
-        send_live_edition
-        send_draft_edition
+      republishing_actions.each do |action|
+        if action[:edition_is_argument]
+          send(action[:method], action[:edition])
+        else
+          send(action[:method])
+        end
       end
     end
   end
 
+  def preview(document_id)
+    set_document_and_editions(document_id)
+
+    return [] unless the_document_has_non_superseded_editions_to_republish?
+
+    republishing_actions.reject { |action| action[:method] == :patch_links }
+  end
+
 private
 
+  def set_document_and_editions(document_id)
+    @document = Document.find(document_id)
+
+    # this the latest edition in a visible state ie: withdrawn, published
+    @live_edition = document.live_edition
+
+    @pre_publication_edition = document.pre_publication_edition
+    @latest_unpublished_edition = document.editions.unpublished.last
+  end
+
+  def republishing_actions
+    if the_document_has_been_unpublished?
+      actions = [
+        { method: :send_unpublish, edition: latest_unpublished_edition, status: "Unpublished", edition_is_argument: true },
+      ]
+
+      actions << { method: :send_draft_edition, edition: pre_publication_edition, status: "Draft" } if pre_publication_edition.present?
+
+      actions
+    elsif the_document_has_been_withdrawn?
+      [{ method: :send_published_and_withdraw, edition: live_edition, status: "Live, withdrawn" }]
+    elsif there_is_only_a_draft?
+      [
+        { method: :patch_links },
+        { method: :send_draft_edition, edition: pre_publication_edition, status: "Draft" },
+      ]
+    elsif there_is_only_a_live_edition?
+      [
+        { method: :patch_links },
+        { method: :send_live_edition, edition: live_edition, status: "Live" },
+      ]
+    elsif there_is_a_newer_draft?
+      [
+        { method: :patch_links },
+        { method: :send_live_edition, edition: live_edition, status: "Live" },
+        { method: :send_draft_edition, edition: pre_publication_edition, status: "Draft" },
+      ]
+    end
+  end
```

Comment:
Not sure how I feel about this pattern/methods that return (arrays of) these kinds of custom meta-ish-data objects. The `status` key is also very much a presentational detail, which might be out of place in a worker, but I suppose this is a `#preview` method

It works, but I'm definitely open to other suggestions for how to go about fetching a list of editions for playing back to the user

I did consider having an entirely separate method returning this information, rather than abstracting into this actions methods used in both `#perform` and `#preview`, but then there's much greater risk of them getting out of sync and previews being inaccurate